utilz/map_update.rb

```diff
@@ -0,0 +1,663 @@
+
+    @options[:geocode_projects] = "#{@options[:geocode_projects]}/" unless @options[:geocode_projects][-1] == '/'
+    p "Looking for MD files in #{@options[:geocode_projects]}"
+    all_filez = Dir.glob("#{@options[:geocode_projects]}**/*.md").select{ |e| File.file? e }
+    len = all_filez.length
+    idx = 0
+    all_filez.each do |file|
+      status_update(len:len, idx:idx)
+      project = read_md file: file
+
+      # find the place tag
+      if project[:yml]["place"] != ""
+        theplace = project[:yml]["place"]
+      else
+        theplace = ""
+      end
+
+      redis = Redis.new(host: "localhost")
+
+      # rest for a second so the requests don't come too fast
+      # for location iq, needs to be less than 2 requests per second
+      # for google, needs to be less than 50 requests per second
+      sleep(0.25)
+
+      @client = GooglePlaces::Client.new("AIzaSyCGucx-_XZJLn-Gd1cHme18vk6osePAE3w")
+
+      if redis.get(theplace) != nil
+        p "cached"
+        coords = redis.get(theplace).delete('[]').split
+        location = coords
+      else
+        if theplace != ""
+          api_call = @client.spots_by_query(theplace)
+          location = nil
+        else
+          p "something wrong with: "+theplace
+          p "try something else?"
+          new_add = gets.chomp
+          api_call = @client.spots_by_query(new_add)
+        end
+      end
+
+      while location == nil
+        # if there's an error in the look up
+        while api_call == []
+          p theplace
+          p "ERROR - can't geocode this location, try another address?"
+          new_add = gets.chomp
+          api_call = @client.spots_by_query(new_add)
+        end
+
+        # if there is more than one location
+        if api_call.count > 1
+          p theplace
+          p "choose from the following by entering 'y' or 'n' (or to enter another address, 'a'): "
+          for n in api_call
+            begin
+              puts "is this it?: #{n.name.to_s} at #{n.formatted_address.to_s}"
+            rescue
+              puts "is this it?: #{n}"
+            end
+            choice = gets.chomp
+            if choice == "y"
+              location = n
+              break
+            end
+            if choice == "a"
+              api_call = []
+              break
+            end
+          end
+        else
+          location = api_call[0]
+        end
+      end
+
+      #p location
+      if redis.get(theplace) != nil
+        lat = coords[0]
+        long = coords[1]
+      else
+        lat = location.lat
+        long = location.lng
+      end
+      if @options[:cache_it] and redis.get(theplace) == nil
+        redis.set(theplace, [lat, long])
+        p "cached it!"
+      end
+      p lat.to_s+" "+long.to_s
+
+      title = project[:yml]["title"]
+      year = project[:yml]["volume"].to_s
+      pagenums = project[:yml]["pages"].to_s
+      link = "\/volume\/"+year+"#"+year+"-"+pagenums
```

Comment:
```suggestion
      link = "/volume/#{year}##{year}-#{pagenums}"
```
so this is pretty minor but i'd like to encourage you to avoid using the `+` operator to concatenate strings. using string interpolation `"nan nan nan #{watman}"` is much preferred because it doesn't get weird when you have types other than string. uh here's a pretty short and pretty funny video describing this: https://www.destroyallsoftware.com/talks/wat